### ISSUE TITLE
Support for policy exemption

### ISSUE THREAD

**Issue by beandrad** [hashicorp/terraform-provider-azurerm#9197]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add a resource to create policy exemptions. 

### New or Affected Resource(s)

* azurerm_policy_exemption

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
resource "azurerm_policy_exemption" "example" {
  name                 = "AllowExpensiveVM"
  scope                = "/subscriptions/ae640e6b-ba3e-4256-9d62-2993eecfa6f2/resourceGroups/demoCluster"
  exemption_category   = "Waiver"
  display_name         = "Exempt demo cluster"
  description          = "Exempt demo cluster from limit sku"
  expires_on           = "2050-01-01T00:00:00"
  policy_assignment_id = "/subscriptions/ae640e6b-ba3e-4256-9d62-2993eecfa6f2/providers/Microsoft.Authorization/policyAssignments/CostManagement"
  policy_definition_reference_ids = [
    "Limit_Skus"
  ]
}
```

### References

The corresponding Rest API endpoints are described [here](https://docs.microsoft.com/en-us/rest/api/resources/policyexemptions). 

The client in the Azure Go SDK can be found [here](https://github.com/Azure/azure-sdk-for-go/blob/e6eee770c9ea598721defb0b1c12cecc9d67d77c/services/preview/resources/mgmt/2020-03-01-preview/policy/policyapi/interfaces.go#L85).

**Comment by ekini** [comment]
What happened to #9392?

**Comment by ArcturusZhang** [comment]
> What happened to #9392?

Hi @ekini we need to ensure that this resource would not receive breaking changes (at least big ones) since it comes from a preview api-version. I will reopen the PR once we get a proper reply from the corresponding Azure service team.

**Comment by VinceGall** [comment]
Also interested in this feature.

**Comment by kmcquade** [comment]
> I will reopen the PR once we get a proper reply from the corresponding Azure service team.
@ArcturusZhang - did you get a reply from the Azure service team? Just wanted to follow up.

**Comment by ArcturusZhang** [comment]
> > I will reopen the PR once we get a proper reply from the corresponding Azure service team.
> > @ArcturusZhang - did you get a reply from the Azure service team? Just wanted to follow up.

Sorry I do not have any news to share about this resource right now. But I will put new comments here once I get any updates, thank you

**Comment by kmcquade** [comment]
@ArcturusZhang - I'd be happy to also raise it with our Microsoft contacts as well to see if we can get any updates from them. I will ask them that question you mentioned above and see if that helps.

**Comment by danielvca** [comment]
I`m also keen on this one... Need to implement policy exemptions to my TF project

**Comment by ArcturusZhang** [comment]
> I`m also keen on this one... Need to implement policy exemptions to my TF project

We are waiting for the GA announcement of this new resource now.

**Comment by danielvca** [comment]
> > I`m also keen on this one... Need to implement policy exemptions to my TF project
> 
> We are waiting for the GA announcement of this new resource now.

Do you know if there's an ETA? Im implementing a workaround using poweshell and local-exec in TF... Not the best solution, but it`s kinda urgent for me to get this deplyed.

**Comment by ArcturusZhang** [comment]
Well, unfortunately I do not have an ETA for this right now... sorry about that...

**Comment by danielvca** [comment]
> Well, unfortunately I do not have an ETA for this right now... sorry about that...

i noticed you closed your PR  (#9392 ). Any reasons for that? 

Also, thaks for replying to my comment.

**Comment by ArcturusZhang** [comment]
> > Well, unfortunately I do not have an ETA for this right now... sorry about that...
> 
> i noticed you closed your PR (#9392 ). Any reasons for that?
> 
> Also, thaks for replying to my comment.

I have to close it because we are planning to have it when it becomes GA

**Comment by jesseloudon** [comment]
Meanwhile... have released a module which leverages Terraform's [resource group template deployment](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group_template_deployment) for managing [policy exemptions](https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/policyexemptions) via ARM template

https://registry.terraform.io/modules/globalbao/policy-exemptions/azurerm/latest

**Comment by qrilka** [comment]
The link to Rest API endpoint in the description should point to https://docs.microsoft.com/en-us/rest/api/policy/policy-exemptions (the current URL gives 404)

**Comment by stefan-rapp** [comment]
Any updates? Need it for customer project. Thx

**Comment by richardlock** [comment]
Microsoft Support confirmed that Policy Exemptions are generally available.

**Comment by ernest97** [comment]
Any updates?

**Comment by karishma-kohli** [comment]
I need to implement Policy exemptions in my project so I am very interested in this feature. Can anyone please tell if it's GA now?

**Comment by sean-keane25** [comment]
It appears the code changes for this have completed and a PR was open but it was never merged ?

https://github.com/hashicorp/terraform-provider-azurerm/pull/9392

**Comment by bubbletroubles** [comment]
This feature is now in GA - see confirmation via this commit from the MS product group

https://github.com/MicrosoftDocs/azure-docs/commit/c6c17a79013d57e11558abac4f804088f180c9bb

**Comment by karishma-kohli** [comment]
I still dont see it in azurerm, so not GA yet I guess.

**Comment by bubbletroubles** [comment]
The capability is now GA from Microsoft - there is no Terraform AzureRM resource yet (previously it was waiting for the Microsoft capability to go GA, that has now  happened, so this should be able to progress.

**Comment by karishma-kohli** [comment]
Ah ok!!! Waiting for them to add this capability.

**Comment by karishma-kohli** [comment]
@[beandrad](https://github.com/beandrad) any info on when would policy exemptions be included in azurerm?

**Comment by mbfrahry** [comment]
This was just merged in #16293 and should go out in v3.2.0 of the provider!

**Comment by karishma-kohli** [comment]
That's great news. Thanks for the update @mbfrahry
